modules/core/src/main/java/org/apache/ignite/internal/processors/query/schema/SchemaOperationWorker.java

```diff
@@ -95,7 +96,7 @@ public SchemaOperationWorker(GridKernalContext ctx, GridQueryProcessor qryProc,
 
         if (err != null)
             fut.onDone(err);
-        else if (nop || !cacheRegistered)
+        else if (nop || !(cacheRegistered || op instanceof SchemaAddQueryEntityOperation))
```

Comment:
Can we rewrite this statement to:
```
(nop || (!cacheRegistered && !(op instanceof SchemaAddQueryEntityOperation)))
```
We want to say here: Complete the future if 
    1. There is no operation OR
    2. Cache not registered AND operation is not `SchemaAddQueryEntityOperation`